.github/workflows/release.yml

```diff
@@ -1,45 +1,15 @@
-name: Release Binary Package
+# Cut a release whenever a new tag is pushed to the repo.
+# You should use an annotated tag, like `git tag -a v1.2.3`
+# and put the release notes into the commit message for the tag.
+name: Release
+
 on:
   push:
     tags:
-      - v*
+      - "v*.*.*"
 
 jobs:
-  generate:
-    name: Create release-artifacts
-    runs-on: ubuntu-latest
-    steps:
-      - name: "Checkout the sources"
-        uses: actions/checkout@v2.3.1
-      - name: Install JDK 11
-        uses: actions/setup-java@v3
-        with:
-          distribution: "zulu"
-          java-version: "11"
-      - name: Setup Bazelisk
-        uses: bazelbuild/setup-bazelisk@v1
-      - name: Mount bazel cache
-        uses: actions/cache@v2
-        with:
-          path: "/home/runner/.cache/bazel"
-          key: caches-${{ runner.os }}-release
-      - name: Build release artifact
-        run: bazelisk build //:rules_kotlin_release
-      - name: Create release sha256
-        run: shasum -a 256 bazel-bin/rules_kotlin_release.tgz > bazel-bin/rules_kotlin_release.tgz.sha256
-      - name: Upload bazel-bin/rules_kotlin_release.tgz
-        uses: svenstaro/upload-release-action@2.2.1
-        with:
-          repo_token: ${{ secrets.GITHUB_TOKEN }}
-          file: bazel-bin/rules_kotlin_release.tgz
-          asset_name: rules_kotlin_release.tgz
-          tag: ${{ github.ref }}
-          overwrite: true
-      - name: Upload bazel-bin/rules_kotlin_release.tgz.sha256
-        uses: svenstaro/upload-release-action@2.2.1
-        with:
-          repo_token: ${{ secrets.GITHUB_TOKEN }}
-          file: bazel-bin/rules_kotlin_release.tgz.sha256
-          asset_name: rules_kotlin_release.tgz.sha256
-          tag: ${{ github.ref }}
-          overwrite: true
+  release:
+    uses: bazel-contrib/.github/.github/workflows/release_ruleset.yaml@v4
```

Comment:
This changes howe we cut releases in `rules_kotlin`. The way to create releases is to now push a tag like `v4.0`